### DISCUSSION TITLE
Load __NEXT_DATA__ without using inline javascript for ISR

### DISCUSSION THREAD

**Discussion by meghint** [vercel/next.js#67801] [Help]
### Summary

Hello everyone,

I've been working on a Next.js project where we use Incremental Static Regeneration (ISR) to improve our site's performance and SEO. However, I've encountered a potential security concern with the default way Next.js injects __NEXT_DATA__ as an inline JavaScript script in the HTML. I'm looking for a way to load __NEXT_DATA__ without relying on inline JavaScript.

**Here are the details of my setup:**

Current Configuration: Standard ISR setup as per the Next.js documentation

**What I want to achieve:**

Eliminate the use of inline JavaScript for injecting __NEXT_DATA__.
Ensure the __NEXT_DATA__ is still available for hydration and proper functioning of the ISR.

**Questions:**

- Is it possible to load __NEXT_DATA__ externally rather than embedding it directly in the HTML?
- 
- If yes, what changes would be required in the Next.js configuration or the ISR setup to achieve this?
- 
- Are there any potential downsides or performance implications of loading __NEXT_DATA__ externally?
- 
- Has anyone implemented a similar solution, and if so, what were your experiences and challenges?

### Additional information

```js
Next.js Version: 14.x
Deployment Platform: SST / AWS
```

**Comment by icyJoseph** [comment]
Could you model the threat a bit more? What's the potential vector for XSS attack here?

**Reply by meghint** [reply]
> Could you model the threat a bit more? What's the potential vector for XSS attack here?

Our main purpose is to move __NEXT_DATA__ from inline javascript.

**Comment by davidmayr** [comment]
Why would you want that?

If you are worried about users reading that data, you are doing something wrong. You should never attempt to send sensitive information to the client in any way. 

Should you be worried that users might modify this data, you are also doing something wrong. Never ever trust the client side.

Unless I am missing something: There is literally no reason to do, what you want to do.
